velox/functions/sparksql/tests/SplitTest.cpp

```diff
@@ -0,0 +1,282 @@
+    auto expectedResult = makeArrayVector(expected);
+    VectorPtr result;
+    if (limit.has_value()) {
+      auto limits = makeConstant<int32_t>(limit.value(), numRows);
+      auto input = makeRowVector({strings, delims, limits});
+      result = evaluate("split(c0, c1, c2)", input);
+    } else {
+      auto input = makeRowVector({strings, delims});
+      result = evaluate("split(c0, c1)", input);
+    }
+    assertEqualVectors(expectedResult, result);
+  };
+};
+
+TEST_F(SplitTest, basic) {
+  auto limit = -1;
+  auto delim = ",";
+  auto numRows = 4;
+  auto input = std::vector<std::string>{
+      {"I,he,she,they"}, // Simple
+      {"one,,,four,"}, // Empty strings
+      {"a,\xED,\xA0,123"}, // Not a well-formed UTF-8 string
+      {""}, // The whole string is empty
+  };
+  auto expected = std::vector<std::vector<std::string>>({
+      {"I", "he", "she", "they"},
+      {"one", "", "", "four", ""},
+      {"a", "\xED", "\xA0", "123"},
+      {""},
+  });
+
+  // No limit provided.
+  testSplit(input, delim, std::nullopt, numRows, expected);
+
+  // Limit <= 0.
+  testSplit(input, delim, limit, numRows, expected);
+
+  // High limit, the limit greater than the input string size.
+  limit = 10;
+  testSplit(input, delim, limit, numRows, expected);
+
+  // Small limit, the limit is smaller than or equals to input string size.
+  limit = 3;
+  expected = {
+      {"I", "he", "she,they"},
+      {"one", "", ",four,"},
+      {"a", "\xED", "\xA0,123"},
+      {""},
+  };
+  testSplit(input, delim, limit, numRows, expected);
+
+  // limit = 1, the resulting array only has one entry to contain all input.
+  limit = 1;
+  expected = {
+      {"I,he,she,they"},
+      {"one,,,four,"},
+      {"a,\xED,\xA0,123"},
+      {""},
+  };
+  testSplit(input, delim, limit, numRows, expected);
+
+  // Non-ascii delimiter.
+  delim = "లేదా";
+  input = {
+      {"синяя сливаలేదా赤いトマトలేదా黃苹果లేదాbrown pear"},
+      {"зелёное небоలేదాలేదాలేదా緑の空లేదా"},
+      {"aలేదా\xEDలేదా\xA0లేదా123"},
+      {""},
+  };
+  expected = {
+      {"синяя слива", "赤いトマト", "黃苹果", "brown pear"},
+      {"зелёное небо", "", "", "緑の空", ""},
+      {"a", "\xED", "\xA0", "123"},
+      {""},
+  };
+  testSplit(input, delim, std::nullopt, numRows, expected);
+  limit = -1;
+  testSplit(input, delim, limit, numRows, expected);
+  limit = 10;
+  testSplit(input, delim, limit, numRows, expected);
+  limit = 3;
+  expected = {
+      {"синяя слива", "赤いトマト", "黃苹果లేదాbrown pear"},
+      {"зелёное небо", "", "లేదా緑の空లేదా"},
+      {"a", "\xED", "\xA0లేదా123"},
+      {""},
+  };
+  testSplit(input, delim, limit, numRows, expected);
+  limit = 1;
+  expected = {
+      {"синяя сливаలేదా赤いトマトలేదా黃苹果లేదాbrown pear"},
+      {"зелёное небоలేదాలేదాలేదా緑の空లేదా"},
+      {"aలేదా\xEDలేదా\xA0లేదా123"},
+      {""},
+  };
+  testSplit(input, delim, limit, numRows, expected);
+
+  // Cover case that delimiter not exists in input string.
+  delim = "A";
```

Comment:
Ditto. Might be more readable to hard-coded delim:

```
testSplit(input, "A", std::nullopt, numRows, expected);
```

Reply:
done